### DISCUSSION TITLE
Is somebody able to explain to me why not lazyloading tabout breaks nvim-cmp ? (Lazy)

### DISCUSSION THREAD

**Discussion by bro69noscope** [abecodes/tabout.nvim#73] [Q&A]
Would love to understand the reason. If Tabout doesn't have: `lazy = false` set, then nvim cmp just doesnt work anymore... took a bit of time for me to realize what was wrong...

**Comment by abecodes** [comment]
@avario-cpu No real idea, I am still rocking packer and not using cmp.
Guess there is some overwriting of keybindings happening.
Can you try to lazy load an have a load order set?

**Comment by bro69noscope** [comment]
I think this is just me being a confused noob, there were many implication about the lazy events I didn't fully grasp, such as the event `InsertCharPre` which will trigger when accessing the Telescope pickers I've set up from my dashboard and am used to, but will not trigger when inserting characters in the cmd line (either the `/` one, or the `:` one).

The habit of accessing files using those pickers and noticing the inconsistent behavior of cmp in the `/ search` f.ex. when entering vim with a direct `nvim file.txt` command versus an empty `nvim` command, in my case leading to a dashboard plugin where I would then usually access files via a picker and hence triggering `InsertCharPre`, made it pretty confusing and all i could tell was that setting `lazy = false` on tabout would solve any and all issues related to those inconsistencies.

But, playing around, the events:
`event = "InsertCharPre",` for tabout nvim 
and
`event = { "InsertEnter", "CmdlineEnter" },` for nvim cmp
seemed to be pretty optimal and to work in tandem reliably
